### DISCUSSION TITLE
STM32F7 Discovery Boards and my frustration.

### DISCUSSION THREAD

**Discussion by rvxfahim** [ImpulseAdventure/GUIslice#390] [General]
I would like to know from the community if anyone has successfuly used the built in LCD of f7 discovery boards with Guislice library.

This board STM32F746G comes built with a display which is a 40 pin connection display module RK043FN48H with driver IC OTA5180A. From what I understand, I can probably hook up this board and install the arduino bootloader in it since there is official support for it and install a guislice supported LCD shield manually with pins defined in guislice library and avoid it's built in LCD, but that would be a waste of resource and money. 

Whereas STM32 touchGFX design software can utilize DMA2D also known as chrom accelerator of STM32 and accelerate the display interface of the built in module and finally generate code for the display where the rest of the libraries are built-in CubeMX IDE of ST. 

To be honest touchGFX gui builder has very nice modern templates, but the horror begins when actual MCU coding has to be done in the cubeMX ide where setting up simple PWM and Timers requires one to write millions of code and pre-setup of the cubemx configurator. Please don't bash me, I am an amateur who takes on ambitious projects in atmega2560/328 chips and design industrial machines easily with arduino IDE and the hundreds of libraries available in the internet.

A little rant - why is ST so not maker friendly, they have so many powerful boards but they are so hard to program.

**Comment by Pconti31** [comment]
@rvxfahim It does look like a good value but they seem to only want you to use their software given the lack of documentation.

I can't help you with GUIslice but this link might help you to get Arduino IDE working.
[Arduino_Core_STM32](https://github.com/stm32duino/Arduino_Core_STM32#discovery-boards)
Paul--
